Some installed packages could not be loaded because they contain native modules that were compiled for an earlier version of Atom.

Atom 1.40.0
Windows 10
Hydrogen 2.8.0

I tried to launch Atom, but a red bug appeared with the following message regarding Hydrogen 2.8.0:

```
Some installed packages could not be loaded because they contain native modules that were compiled for an earlier version of Atom.
```
with the following Details:
```
gyp ERR! configure error 
gyp ERR! stack Error: Can't find Python executable "C:\Users\usernumb...\AppData\Local\Programs\Python\Python37-32\python.EXE", you can set the PYTHON env variable.
gyp ERR! stack     at PythonFinder.failNoPython (C:\Users\i0310451\AppData\Local\atom\app-1.40.0\resources\app\apm\node_modules\npm\node_modules\node-gyp\lib\configure.js:492:19)
gyp ERR! stack     at PythonFinder.<anonymous> (C:\Users\usernumb...\AppData\Local\atom\app-1.40.0\resources\app\apm\node_modules\npm\node_modules\node-gyp\lib\configure.js:517:16)
gyp ERR! stack     at C:\Users\usernumb...\AppData\Local\atom\app-1.40.0\resources\app\apm\node_modules\npm\node_modules\graceful-fs\polyfills.js:284:29
gyp ERR! stack     at FSReqWrap.oncomplete (fs.js:182:21)
gyp ERR! System Windows_NT 10.0.17134
gyp ERR! command "C:\\Users\\usernumb...\\AppData\\Local\\atom\\app-1.40.0\\resources\\app\\apm\\bin\\node.exe" "C:\\Users\\usernumb...\\AppData\\Local\\atom\\app-1.40.0\\resources\\app\\apm\\node_modules\\npm\\node_modules\\node-gyp\\bin\\node-gyp.js" "rebuild"
gyp ERR! cwd C:\Users\usernumb...\.atom\packages\Hydrogen\node_modules\zeromq
gyp ERR! node -v v10.2.1
gyp ERR! node-gyp -v v3.7.0
gyp ERR! not ok 
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! zeromq@5.1.0 install: `node scripts/prebuild-install.js || (node scripts/preinstall.js && node-gyp rebuild)`
npm ERR! Exit status 1
npm ERR! 
npm ERR! Failed at the zeromq@5.1.0 install script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     C:\Users\usernumb...\.atom\.apm\_logs\2019-08-13T15_38_58_222Z-debug.log
```

What I tried: 
- I tried several times to "Rebuild Packages", but the same error message occurs.
- As it's my PC at work, and it is behind a firewall, I tried the same procedure but connecting to internet through my cellphone, but it didn't work

I saw 2 similar issures: [#308 ](https://github.com/platformio/platformio-atom-ide-terminal/issues/308), [#11](https://github.com/peterdotjs/atom-synced-sidebar/issues/11), but I could not solve the problem

Okay, so atom/apm is to blame: I experimented the installations of the packages switching atom/apm's version.

> On the current master: Fails as reported
```
C:\Users\aviat\github\atom-pkgs\apm>git checkout master
Switched to branch 'master'
Your branch is up to date with 'origin/master'.

C:\Users\aviat\github\atom-pkgs\apm>bin\npm run build

> atom-package-manager@2.4.2 build C:\Users\aviat\github\atom-pkgs\apm
> node script/check-version.js && grunt

Running "coffee:glob_to_multiple" (coffee) task
>> 42 files created.

Running "coffeelint:src" (coffeelint) task
>> 42 files lint free.

Running "coffeelint:test" (coffeelint) task
>> 26 files lint free.

Running "coffeelint:gruntfile" (coffeelint) task
>> 1 file lint free.

Done.

C:\Users\aviat\github\atom-pkgs\apm>bin\apm --version
apm  2.4.2
npm  6.2.0
node 10.2.1 x64
atom 1.40.0
python 3.7.1
git 2.22.0.windows.1
visual studio

C:\Users\aviat\github\atom-pkgs\apm>bin\apm install julia-client
Installing julia-client to C:\Users\aviat\.atom\packages failed

> @pfitzseb/node-pty-prebuilt@0.8.1-prebuilt3 install C:\Users\aviat\AppData\Local\Temp\apm-install-dir-
119717-20220-o0kut2.8vdy\node_modules\julia-client\node_modules\@pfitzseb\node-pty-prebuilt
> prebuild-install || node scripts/install.js


C:\Users\aviat\AppData\Local\Temp\apm-install-dir-119717-20220-o0kut2.8vdy\node_modules\julia-client\nod
e_modules\@pfitzseb\node-pty-prebuilt>if not defined npm_config_node_gyp (node "C:\Users\aviat\github\at
om-pkgs\apm\node_modules\npm\node_modules\npm-lifecycle\node-gyp-bin\\..\..\node_modules\node-gyp\bin\no
de-gyp.js" rebuild )  else (node "C:\Users\aviat\github\atom-pkgs\apm\node_modules\npm\node_modules\node
-gyp\bin\node-gyp.js" rebuild )

prebuild-install WARN install No prebuilt binaries found (target=3.1.10 runtime=node arch=x64 libc= plat
form=win32)
gyp ERR! configure error
gyp ERR! stack Error: Can't find Python executable "C:\Users\aviat\Anaconda3\python.EXE", you can set th
e PYTHON env variable.
gyp ERR! stack     at PythonFinder.failNoPython (C:\Users\aviat\github\atom-pkgs\apm\node_modules\npm\no
de_modules\node-gyp\lib\configure.js:492:19)
gyp ERR! stack     at PythonFinder.<anonymous> (C:\Users\aviat\github\atom-pkgs\apm\node_modules\npm\nod
e_modules\node-gyp\lib\configure.js:517:16)
gyp ERR! stack     at C:\Users\aviat\github\atom-pkgs\apm\node_modules\npm\node_modules\graceful-fs\poly
fills.js:284:29
gyp ERR! stack     at FSReqWrap.oncomplete (fs.js:182:21)
gyp ERR! System Windows_NT 10.0.18362
gyp ERR! command "C:\\Users\\aviat\\github\\atom-pkgs\\apm\\bin\\node.exe" "C:\\Users\\aviat\\github\\at
om-pkgs\\apm\\node_modules\\npm\\node_modules\\node-gyp\\bin\\node-gyp.js" "rebuild"
gyp ERR! cwd C:\Users\aviat\AppData\Local\Temp\apm-install-dir-119717-20220-o0kut2.8vdy\node_modules\jul
ia-client\node_modules\@pfitzseb\node-pty-prebuilt
gyp ERR! node -v v10.2.1
gyp ERR! node-gyp -v v3.7.0
gyp ERR! not ok
npm WARN enoent ENOENT: no such file or directory, open 'C:\Users\aviat\AppData\Local\Temp\apm-install-d
ir-119717-20220-o0kut2.8vdy\package.json'
npm WARN apm-install-dir-119717-20220-o0kut2.8vdy No description
npm WARN apm-install-dir-119717-20220-o0kut2.8vdy No repository field.
npm WARN apm-install-dir-119717-20220-o0kut2.8vdy No README data
npm WARN apm-install-dir-119717-20220-o0kut2.8vdy No license field.

npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! @pfitzseb/node-pty-prebuilt@0.8.1-prebuilt3 install: `prebuild-install || node scripts/install.
js`
npm ERR! Exit status 1
npm ERR!
npm ERR! Failed at the @pfitzseb/node-pty-prebuilt@0.8.1-prebuilt3 install script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     C:\Users\aviat\.atom\.apm\_logs\2019-08-16T15_06_34_768Z-debug.log
```

> On `2.2.4` branch, which is the last version before atom/apm#845 is introduced: Suceed
```
C:\Users\aviat\github\atom-pkgs\apm> git checkout -b 2.2.4 e446494af555d55fc171166380adf6c8fbd4a989
Switched to a new branch '2.2.4'

C:\Users\aviat\github\atom-pkgs\apm>bin\npm run build

> atom-package-manager@2.2.4 build C:\Users\aviat\github\atom-pkgs\apm
> node script/check-version.js && grunt

Running "coffee:glob_to_multiple" (coffee) task
>> 42 files created.

Running "coffeelint:src" (coffeelint) task
>> 42 files lint free.

Running "coffeelint:test" (coffeelint) task
>> 26 files lint free.

Running "coffeelint:gruntfile" (coffeelint) task
>> 1 file lint free.

Done.

C:\Users\aviat\github\atom-pkgs\apm>bin\apm --version
apm  2.2.4
npm  6.2.0
node 10.2.1 x64
atom 1.40.0
python 3.7.1
git 2.22.0.windows.1
visual studio

C:\Users\aviat\github\atom-pkgs\apm>bin\apm install julia-client
Installing julia-client to C:\Users\aviat\.atom\packages done
```

The same goes for Hydrogen as well.

--- a/apm/package.json
+++ b/apm/package.json
@@ -6,6 +6,6 @@
     "url": "https://github.com/atom/atom.git"
   },
   "dependencies": {
-    "atom-package-manager": "2.4.2"
+    "atom-package-manager": "2.4.3"
   }
 }

